boa_engine/src/vm/opcode/unary_ops/decrement.rs

```diff
@@ -17,11 +17,16 @@ impl Operation for Dec {
 
     fn execute(context: &mut Context<'_>) -> JsResult<ShouldExit> {
         let value = context.vm.pop();
-        match value.to_numeric(context)? {
-            Numeric::Number(number) => context.vm.push(number - 1f64),
-            Numeric::BigInt(bigint) => {
-                context.vm.push(JsBigInt::sub(&bigint, &JsBigInt::one()));
+        match value {
+            JsValue::Integer(number) if number > i32::MIN => {
```

Comment:
I like the idea of this optimization. Just had one question regarding this, was there a reason to not add `JsValue::BigInt` as well?

Reply:
It was more like I couldn't find a reason to add BigInt here. I didn't see any fast path for BigInt when converting it to a property key, and I thought handling BigInt in the next `match` after calling `to_numeric` is not too slow.